### PR TITLE
Force API updates

### PR DESCRIPTION
**Pull request by edantec**
This PR essentially aims at simplifying the API for force-related cost functions to make it more intuitive.

### CODE REVIEW

File: include/aligator/modelling/contact-map.hpp [hunk 1]
```diff
@@ -32,25 +36,58 @@ template <typename _Scalar> struct ContactMapTpl {
...
     }
   }
 
+  const std::vector<std::string> &getContactNames() const {
+    return contact_names_;
+  }
+
   const std::vector<bool> &getContactStates() const { return contact_states_; }
 
   bool getContactState(const std::size_t i) const { return contact_states_[i]; }
```
Comment (ManifoldFR) [inline include/aligator/modelling/contact-map.hpp:52]: Doing getters here is quite abusive I think. There is no need for this, setting these members as public is enough even if God forbid the user decides to misuse and push_back() elements onto them

File: include/aligator/modelling/contact-map.hpp [hunk 2]
```diff
@@ -32,25 +36,58 @@ template <typename _Scalar> struct ContactMapTpl {
...
+    }
+
+    return contact_states_[id - contact_names_.begin()];
+  }
+
   const PoseVec &getContactPoses() const { return contact_poses_; }
 
   const Vector3s &getContactPose(const std::size_t i) const {
     return contact_poses_[i];
   }
```
Comment (ManifoldFR) [inline include/aligator/modelling/contact-map.hpp:67]: Same remark as above regarding the abuse of getters.

File: include/aligator/modelling/multibody/contact-force.hpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ template <typename Scalar> struct ContactForceDataTpl;
  */
 
 template <typename _Scalar>
-struct ContactForceResidualTpl : StageFunctionTpl<_Scalar>, frame_api {
```
Comment (ManifoldFR) [inline include/aligator/modelling/multibody/contact-force.hpp:21]: Is the `frame_api` mixin struct still useful after these changes? If not, please mark it as deprecated.

File: include/aligator/modelling/multibody/dcm-position.hpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  PinData pin_data_;
+  /// Jacobian of the error
+  typename math_types<Scalar>::Matrix3Xs fJf_;
+
+  DCMPositionDataTpl(const DCMPositionResidualTpl<Scalar> &model);
+};
+
+} // namespace aligator
+
+#include "aligator/modelling/multibody/dcm-position.hxx"
```
Comment (ManifoldFR) [inline include/aligator/modelling/multibody/dcm-position.hpp:63]: No need to include the hxx file here. Just in the cpp file.
```suggestion
```

File: src/modelling/multibody/dcm-position.cpp [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#include "aligator/modelling/multibody/dcm-position.hpp"
```
Comment (ManifoldFR) [inline src/modelling/multibody/dcm-position.cpp:1]:
```suggestion
#include "aligator/modelling/multibody/dcm-position.hxx"
```

File: src/modelling/centroidal/centroidal-wrench-cone.cpp [hunk 6]
```diff
@@ -0,0 +1,8 @@
+#include "aligator/modelling/centroidal/centroidal-wrench-cone.hpp"
```
Comment (ManifoldFR) [inline src/modelling/centroidal/centroidal-wrench-cone.cpp:1]:
```suggestion
#include "aligator/modelling/centroidal/centroidal-wrench-cone.hxx"
```

File: include/aligator/modelling/centroidal/centroidal-wrench-cone.hpp [hunk 7]
```diff
@@ -52,17 +52,20 @@ struct WrenchConeResidualTpl : StageFunctionTpl<_Scalar> {
...
   Eigen::Matrix<Scalar, 17, 6> Jtemp_;
 
-  WrenchConeDataTpl(const WrenchConeResidualTpl<Scalar> *model);
+  CentroidalWrenchConeDataTpl(
+      const CentroidalWrenchConeResidualTpl<Scalar> *model);
 };
 
 } // namespace aligator
 
+#include "aligator/modelling/centroidal/centroidal-wrench-cone.hxx"
```
Comment (ManifoldFR) [inline include/aligator/modelling/centroidal/centroidal-wrench-cone.hpp:67]: No need to include the implementation hxx file here, just in the corresponding cpp file.
```suggestion
```

**Review by ManifoldFR — CHANGES_REQUESTED** [review]
Thanks for these changes @edantec. I did a quick review.

File: include/aligator/modelling/contact-map.hpp [hunk 8]
```diff
@@ -32,25 +36,58 @@ template <typename _Scalar> struct ContactMapTpl {
...
 
+  const std::vector<std::string> &getContactNames() const {
+    return contact_names_;
+  }
+
   const std::vector<bool> &getContactStates() const { return contact_states_; }
 
   bool getContactState(const std::size_t i) const { return contact_states_[i]; }
 
+  bool getContactState(const std::string &name) const {
```
[the remark this answers is not in the document]
Comment (edantec) [reply include/aligator/modelling/contact-map.hpp:46]: getContactState by size_t has been erased, only getContactState by name remains. All vectors are now public.

**Review by ManifoldFR — APPROVED** [review]

**Comment by ManifoldFR** [comment]
Congratulations @edantec and thanks for this PR

**Pull request merged**